File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/library/libraryFilterRequest.ts
import { PaginatedRequest as PaginationRequest } from "../../..";

export interface LibraryFilterRequest extends PaginationRequest {
    containsName: string;
}

export function defaultLibraryFilterRequest(): LibraryFilterRequest {
    return {
        containsName: "",
        pageNumber: 0,
        pageSize: 0,
    };
}

File: src/ELibrary.Backend/LibraryShopEntities/Filters/LibraryFilterRequest.cs
﻿using Pagination;

namespace LibraryShopEntities.Filters
{
    public class LibraryFilterRequest : PaginationRequest
    {
        public string ContainsName { get; set; } = string.Empty;

        public override string ToString()
        {
            return $"PageNumber: {PageNumber}, PageSize: {PageSize}, ContainsName: '{ContainsName}'";
        }
    }
}
